pinot-query-planner/src/main/java/org/apache/calcite/rel/hint/PinotHintOptions.java

```diff
@@ -29,12 +29,28 @@ public class PinotHintOptions {
   public static final String AGGREGATE_HINT_OPTIONS = "aggOptions";
   public static final String JOIN_HINT_OPTIONS = "joinOptions";
 
+  /**
+   * Hint to denote that the aggregation node is the final aggregation stage which extracts the final result.
+   */
+  public static final String INTERNAL_AGG_OPTIONS = "aggOptionsInternal";
+
   private PinotHintOptions() {
     // do not instantiate.
   }
 
+  public static class InternalAggregateOptions {
+    public static final String AGG_TYPE = "agg_type";
+    public enum AggType {
```

Comment:
suggest add boolean methods like: isFinal(), isLeaf()... for comparison simplicity.

Reply:
will do in the next iteration. having boolean method is not that useful IMO b/c there's still 4 combination